Integrations/ThreatQ/ThreatQ.py

```diff
@@ -0,0 +1,1248 @@
+        ret['Score'] = dbot_score_map[ind_score]
+
+    return ret
+
+
+def get_tq_score_from_response(score_data):
+    if score_data is None:
+        return None
+    if isinstance(score_data, dict):
+        # score will be max(gen_score, manual_score)
+        gen_score = str(score_data.get('generated_score'))
+        manual_score = score_data.get('manual_score', 0.0)
+        if manual_score is None:
+            manual_score = -1
+        return max(float(gen_score), float(manual_score))
+    else:
+        # score is already defined as a number
+        return float(score_data)
+
+
+def clean_html_from_string(raw_html):
+    """ This function receives an HTML string of a text, and retrieves a clean string of its content.
+
+    Args:
+        raw_html: An HTML format text
+
+    Returns:
+        (string). A clean text string
+    """
+    if not raw_html:
+        return None
+    clean_r = re.compile('<.*?>')
+    clean_text = re.sub(clean_r, '', raw_html)
+    return clean_text
+
+
+def sources_to_request_format(sources):
+    if not sources:
+        return []
+    if isinstance(sources, str):
+        sources = sources.split(',')
+    return [{'name': source} for source in sources]
+
+
+def sources_to_demisto_format(lst):
+    if lst is None:
+        return None
+    return [{
+        'Name': elem.get('name'),
+        'ID': elem.get('pivot', {}).get('id')
+    } for elem in lst]
+
+
+def attributes_to_request_format(attr_names_lst, attr_values_lst):
+    if not attr_names_lst and not attr_values_lst:
+        return []
+    if isinstance(attr_names_lst, str):
+        attr_names_lst = attr_names_lst.split(',')
+    if isinstance(attr_values_lst, str):
+        attr_values_lst = attr_values_lst.split(',')
+    if not attr_names_lst or not attr_values_lst or len(attr_names_lst) != len(attr_values_lst):
+        return_error('Arguments attr_names_lst and attr_values_lst should have the same length.')
+
+    return [{'name': name, 'value': val} for name, val in zip(attr_names_lst, attr_values_lst)]
+
+
+def attributes_to_demisto_format(lst):
+    if lst is None:
+        return None
+    return [{
+        'Name': elem.get('name'),
+        'Value': elem.get('value'),
+        'ID': elem.get('id')
+    } for elem in lst]
+
+
+def content_type_to_demisto_format(c_type_id):
+    # content_type is a file object property
+    return 'text/plain' if c_type_id == 1 else 'text/rtf'
+
+
+def malware_locked_to_request_format(state):
+    # malware_locked is a file object property
+    if not state:
+        return None
+    return 1 if state == 'on' else 0
+
+
+def malware_locked_to_demisto_format(state):
+    return 'on' if state == 1 else 'off'
+
+
+def parse_date(text):
+    valid_formats = ['%Y-%m-%d %H:%M:%S']
+    for fmt in valid_formats:
+        try:
+            return str(datetime.strptime(text, fmt))
+        except ValueError:
+            pass
+    return_error('Time data \'{0}\' does not match any valid format.'.format(text))
```

Comment:
mention which format is valid, and why isn't that in the except?

Reply:
let me know if you think it should be changed after my explanation